### ISSUE TITLE
Integrity constraint violation: duplicate entry

### ISSUE THREAD

**Issue by Seferi** [owncloud/music#748]
Hi, I’m running NCP docker container on a Raspberry pi. I found the server in maintenance mode today, it says Music app needs an update but when I press update, it fails. I tried updating from NCP web panel and with occ command with docker exec. All fails. This is what I’m getting:

Cannot load Zend OPcache - it was already loaded
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Updating …
Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing ‘UPDATE oc_music_albums SET hash = ? WHERE id = ?’ with params [“ea9a343e944251f0130d99602272e1e2”, “14689”]:

SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry ‘-ea9a343e944251f0130d99602272e1e2’ for key ‘ma_user_id_hash_idx’
Update failed
Maintenance mode is kept active
Reset log level

I appreciate the help. Thanks!

P.S. Here is the entry on NC forum:
https://help.nextcloud.com/t/music-app-update-fails-ncp-stays-in-maintenance-mode/74968

**Comment by paulijar** [comment]
Thanks for the report and sorry for the trouble. 

Apparently, my database schema migration script can't handle your music library, but it may be difficult to tell, what actually is the triggering issue here. I have some ideas about special cases, which might cause a problem, though. But maybe there are cases, which I can't even imagine, so I might need to figure out a way to carry on the installation even if the DB migration runs into problems.

As first aid, you could try to remove the album causing the problem directly from the database:
`DELETE FROM oc_music_albums WHERE hash='ea9a343e944251f0130d99602272e1e2';`

After this, the installation should either succeed, or fail to the next similar problem case. If there are lot of such problem cases, then you might prefer to erase your whole music library:
```
DELETE FROM oc_music_albums;
DELETE FROM oc_music_tracks
DELETE FROM oc_music_artists;
DELETE FROM oc_music_playlists;
```
After deleting one album or the whole library, you should be prompted to rescan them the next time you open the Music app.

But even before deleting anything, it would be helpful, if you could check and report the contents of those involved albums. So could you run the following queries:
```
SELECT * FROM oc_music_albums WHERE hash='ea9a343e944251f0130d99602272e1e2';
SELECT * FROM oc_music_albums WHERE id='14689';
```

**Comment by Seferi** [comment]
Thanks for the reply mate. I have never used sql queries. What is the complete command I have to use? Before DELETE FROM , like 'sudo sql DELETE FROM' ?

**Comment by paulijar** [comment]
I now released Music v0.13.2 which should hopefully fix this issue.
